### PR TITLE
Fixed #33132 -- Fixed test client handling of querystring only redirects.

### PR DESCRIPTION
**Pull request by jaap3**
Fixed ticket-33132

Regression in 1e5aa8e1c79252cc810af21294a6e945d11d37b3.

### CODE REVIEW

**Comment by carltongibson** [comment]
Thanks for the quick PR @jaap3! I shall take a look this afternoon. 👍

**Review by carltongibson — APPROVED** [review]
Hi @jaap3 — thanks for this. 

I just pushed a suggested edit in a53566c4c2533fd1371c010749af1c6ba6a72bbb. Let me know what you think, and if you're happy we can squash down and commit. 

👍

File: django/test/client.py [hunk 1]
```diff
@@ -835,8 +835,11 @@ def _handle_redirects(self, response, data='', content_type='', **extra):
             if url.port:
                 extra['SERVER_PORT'] = str(url.port)
 
+            path = url.path
+            # RFC 2616: bare domains without path are treated as the root.
+            if not path and url.netloc:
+                path = '/'
```
Comment (carltongibson) [inline django/test/client.py:841]: I thought this `path = url.path or url.netloc and '/'` was too hard to reason about. 

Then moving about relative to the old comment read clearer I thought... 

🤔

Comment (jaap3) [reply django/test/client.py:841]: Much better, thanks!

Comment (carltongibson) [reply django/test/client.py:841]: OK, thanks. I'll squash it down then. 😄

**Pull request merged**